src/arrayMethodJoin.js

```diff
@@ -4,9 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        str += this[i];
+      }
+
+      if (i !== this.length - 1) {
```

Comment:
Instead of checking if the current iteration is not the last, you can simplify your code by adding the separator after each iteration, and then remove the last separator at the end.